addons/nuxeo-template-rendering/nuxeo-template-rendering-core/src/main/java/org/nuxeo/template/processors/AbstractBindingResolver.java

```diff
@@ -70,106 +78,179 @@ protected DocumentObjectWrapper getWrapper() {
+            case DateValue:
+                return param.getDateValue();
+            case StringValue:
+                return param.getStringValue();
+            case MapValue:
+                Map<String, Object> resultMap = new HashMap<>();
+                Map<String, TemplateInput> map = param.getMapValue();
+                for (String paramName : map.keySet()) {
+                    try {
+                        resultMap.put(paramName, extractValueFromParam(templateBasedDocument, map.get(paramName)));
+                    } catch (NoValueToAddInContext | ValueNotFound e) {
+                        log.debug("Skip param to add " + paramName + " in " + param.getName(), e);
                     }
-                    Property property = null;
+                }
+                return resultMap;
+            case ListValue:
+                List<Object> resultList = new ArrayList<>();
+                for (TemplateInput childParam : param.getListValue()) {
                     try {
-                        property = templateBasedDocument.getAdaptedDoc().getProperty(param.getSource());
-                    } catch (PropertyException e) {
-                        log.warn("Unable to ready property " + param.getSource(), e);
+                        resultList.add(extractValueFromParam(templateBasedDocument, childParam));
+                    } catch (NoValueToAddInContext | ValueNotFound e) {
+                        log.debug("Skip param to add " + childParam.getName() + " in " + param.getName(), e);
                     }
-
-                    Serializable value = null;
-                    if (property != null) {
-                        value = property.getValue();
+                }
+                return resultList;
+            case Content:
+                ContentInputType contentInput = ContentInputType.getByValue(param.getSource());
+                if (BlobContent.equals(contentInput))
+                    return extractBlobContent(doc, param);
+                else if (HtmlPreview.equals(contentInput))
+                    return extractHTMLPreview(doc, param);
+                else {
+                    Serializable docPropertyValue = getDocPropertyValue(doc, propKey);
+                    if (docPropertyValue instanceof String)
+                        return handleHtmlField(param.getName(), (String) getDocPropertyValue(doc, propKey));
+                }
+            case PictureProperty:
+                try {
+                    Serializable docPropertyValue = getDocPropertyValue(doc, propKey);
+                    if (isBlob(docPropertyValue)) {
+                        Blob blob = (Blob) getDocPropertyValue(doc, propKey);
+                        addDefaultMimetypeIfRequired(blob);
+                        return handlePictureField(param.getName(), blob);
                     }
+                } catch (ValueNotFound e) {
+                    return handlePictureField(param.getName(), null);
+                }
```

Comment:
Missing `break` too.

Reply:
This is normal : if the value is not the blob we go through the default behaviour. That's not my choice, it was like that before me.